cmd/cloud-controller-manager/controller-manager.go

```diff
@@ -51,6 +54,7 @@ func main() {
 	defer logs.FlushLogs()
 
 	if err := command.Execute(); err != nil {
-		os.Exit(1)
+		exitCode = 1
+		return
 	}
```

Comment:
Yes, It is just a small change.

flush log before exit seems reasonable logically, but I am not sure if there are useful logs to be flushed. I will check again.

Thanks.

Reply:
Personally when a process "crashes", I find the last things to be logged the most interesting. So I at least am a fan of trying to flush the logs on the way out.

Reply:
I think flush before exit is necessary because the default klog flush interval is 5s, according to
`vendor/k8s.io/component-base/logs/logs.go:57`, which may lose logs.

There are 7 components to change, should we change it all in current mr ?

Reply:
The more components in the PR, the harder it is get all the sign-offs before your PR is stale. 
I would suggest merging this and doing the other components as a follow up.